pkg/agent/factsopenshift.go

```diff
@@ -0,0 +1,110 @@
+package agent
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+type OpenshiftVersionDesired struct {
+	Version string
+}
+
+type OpenshiftVersionHistory struct {
+	State          string
+	Verified       bool
+	Version        string
+	CompletionTime time.Time
+}
+
+type OpenshiftVersionStatus struct {
+	Desired OpenshiftVersionDesired
+	History []OpenshiftVersionHistory
+}
+type OpenshiftVersion struct {
+	Status OpenshiftVersionStatus
+}
+
+func (col factCollector) fetchOpenshiftVersion(ctx context.Context) (*OpenshiftVersionFact, error) {
+	body, err := col.client.RESTClient().Get().AbsPath("/apis/config.openshift.io/v1/clusterversions/version").Do(ctx).Raw()
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// API server doesn't know `clusterversions` or there is no resource, so we are not running on openshift.
+			return nil, nil
+		}
+		return nil, err
+	}
+	var version OpenshiftVersion
+	err = json.Unmarshal(body, &version)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse the openshift version: %w", err)
+	}
+
+	return processOpenshiftVersion(version)
+}
+
+type SemanticVersion struct {
+	Major string
+	Minor string
+	Patch string
+}
+
+type OpenshiftVersionFact struct {
+	Version        SemanticVersion
+	DesiredVersion SemanticVersion
+}
+
+func processOpenshiftVersion(v OpenshiftVersion) (*OpenshiftVersionFact, error) {
+	currentVersion := ""
+	lastedUpdate := time.Time{}
```

Comment:
Is `lasted` a typo of `last` or does this have a special meaning which I don't understand?

Semantically this should be something like (my suggestion is probably a bit long :see_no_evil: )

```suggestion
	mostRecentUpdateCompletionTime := time.Time{}
```

Reply:
Yes that should have been `last` :sweat_smile:.  I think `lastUpdate` should be descriptive enough, especially since the variable is very local.